### DISCUSSION TITLE
Rules with content

### DISCUSSION THREAD

**Discussion by Thomer73** [Security-Onion-Solutions/securityonion#5444] [Unsupported Versions]
Hi everyone,

i don't understand how to create rules with content filtering. 
i try rules like these
**pass tcp any any -> any 445 (msg:"ET POLICY SMB2 NT Create AndX Request For a .bat File"; flow:established,to_server;** content:".W.i.n.A.P.P.S";nocase; sid:1000029; rev:1;)
to pass alerts from our x:\winapps\... directory.

i tried everything. rules are written in all.rules but i receive alerts if:

_network.data.decoded | .....SMB@...................................E..4o...9...7*..C.t.X.^.9......................................... .x.F.....0...P.r.o.g.\.W.i.n.A.P.P.S.\.t.b.\.F.r.e.i.s.c.h.a.l.t.u.n.g.T.B...e.x.e...................MxAc....................QFid....
-- | --_

i tried it with offset ,... but it should not work.
everytime the rule:
_"alert tcp any any -> $HOME_NET 445 (msg:\"ET POLICY SMB2 NT Create AndX Request For an Executable File\"; flow:established,to_server; content:\"SMB\"; depth:8; content:\"|05 00|\"; distance:8; within:2; content:\"|00 2E 00|e|00|x|00|e|00|\"; nocase; distance:0; classtype:bad-unknown; sid:2025701; rev:2; metadata:affected_product Windows_XP_Vista_7_8_10_Server_32_64_Bit, attack_target SMB_Client, created_at 2018_07_16, deployment Perimeter, former_category POLICY, signature_severity Minor, updated_at 2018_07_16;)"_
will take action...

my target is to whitelist some traffic and cleanup alerts in suricata and make it readable. i think writing rules with content filter, is the only way ,or?

**Comment by dougburks** [comment]
Have you checked the documentation?
https://docs.securityonion.net/en/2.3/managing-alerts.html

**Comment by Thomer73** [comment]
I've read the documentation. In principle, I can also create rules and can deactivate certain SIDs. Only the content rules will not work for me.

**Comment by dougburks** [comment]
Looking at your rule, it's possible that your content match `.W.i.n.A.P.P.S"` is not actually matching the content. You might try switching to a hexadecimal content match. 

However, if your X: drive is mapped to your known good file server and it is expected that users would access executable files, then I think that most organizations would simply suppress 2025701 for the IP address of the file server. For more about suppression, please see:
https://docs.securityonion.net/en/2.3/managing-alerts.html
